hw_diag/utilities/network_watchdog.py

```diff
@@ -100,6 +111,44 @@ def restart_network_manager(self) -> None:
         nm_restarted = network_manager_unit.wait_restart()
         self.LOGGER.info(f"Network manager restarted: {nm_restarted}")
 
+    def _send_network_event(self, event_type, action_type, msg) -> None:
+        # don't repeat the events unnecessarily
+        new_fingerprint = event_fingerprint(event_type, action_type, msg)
+        if new_fingerprint == self.last_fingerprint:
+            return
+
+        # save last state
+        self.last_fingerprint = new_fingerprint
+        self.last_event = event_type
+        self.last_action = action_type
+
+        balena_state = system_metrics.get_balena_metrics()
+        # print(json.dumps(balena_state, indent=4))
+        network_stats = system_metrics.get_network_statistics()
+        event = {
+            EVENT_TYPE_KEY: event_type,
+            'action_type': action_type,
+            'msg': msg,
+            'serial': system_metrics.get_serial_number(),
+            'variant': system_metrics.get_variant(),
+            'firmware_version': system_metrics.get_firmware_version(),
+            'region_override': system_metrics.get_region_override(),
+            'uptime': str(timedelta(seconds=uptime())),
+            'balena_health': system_metrics.are_all_services_up(balena_state),
+            'packet_errors': system_metrics.total_packet_errors(network_stats),
+            'balena_state': json.dumps(balena_state),
```

Comment:
thinking of getting rid of these two json as strings. I initially thought that it might be useful but can't think of a good usecase. For failed services I will add another field "failed_services".

Reply:
these were removed.